packages/terra-compact-interactive-list/src/CompactInteractiveList.jsx

```diff
@@ -99,9 +132,96 @@ const CompactInteractiveList = (props) => {
+    const focusedCellElement = listRef.current.children[row + 1].children[cell];
+    const interactiveChildren = getFocusableElements(focusedCellElement);
+    if (interactiveChildren?.length > 0 && interactiveChildren[0]) {
+      // currently a cell can have only one interact-able element in it, which gets auto-focused.
+      interactiveChildren[0].focus();
+    } else {
+      // cell gets focus if there is no interact-able elements in it.
+      focusedCellElement.focus();
+    }
+  };
+
+  const setFocusedCell = ({ rowIndex, columnIndex }) => {
+    focusedCell.current = { row: rowIndex, cell: columnIndex };
+  };
+
+  const handleKeyDown = (event) => {
+    let moveFocusTo = focusedCell.current;
+
+    const targetElement = event.target;
+
+    // Allow default behavior if the event target is an editable field
+
+    if (event.keyCode !== KeyCode.KEY_TAB
+        && (isTextInput(targetElement)
+            || ['textarea', 'select'].indexOf(targetElement.tagName.toLowerCase()) >= 0
+            || (targetElement.hasAttribute('contentEditable') && targetElement.getAttribute('contentEditable') !== false))) {
+      return;
+    }
+
+    const key = event.keyCode;
+    switch (key) {
+      case KeyCode.KEY_UP:
+        moveFocusTo = handleUpKey(focusedCell.current, numberOfColumns, flowHorizontally, rows.length);
+        event.preventDefault();
+        break;
+      case KeyCode.KEY_DOWN:
+        moveFocusTo = handleDownKey(focusedCell.current, numberOfColumns, flowHorizontally, rows.length);
+        event.preventDefault();
+        break;
+      case KeyCode.KEY_LEFT: {
+        moveFocusTo = handleLeftKey(event, focusedCell.current, numberOfColumns, flowHorizontally, columns.length, rows.length);
+        break;
+      }
+      case KeyCode.KEY_RIGHT: {
+        moveFocusTo = handleRightKey(event, focusedCell.current, numberOfColumns, flowHorizontally, columns.length, rows.length);
+        break;
+      }
+      case KeyCode.KEY_HOME:
+        moveFocusTo = handleHomeKey(event, focusedCell.current, numberOfColumns, flowHorizontally, rows.length);
+        break;
+      case KeyCode.KEY_END:
+        moveFocusTo = handleEndKey(event, focusedCell.current, numberOfColumns, flowHorizontally, columns.length, rows.length);
+        break;
+      case KeyCode.KEY_ESCAPE:
+        if (onClearSelection) {
+          onClearSelection();
+        }
+        event.preventDefault();
+        return;
+      case KeyCode.KEY_TAB:
+        moveFocusFromElement(listRef, id, !event.shiftKey);
+        event.preventDefault();
+        return;
+      default:
+        return;
+    }
+    focusedCell.current = moveFocusTo;
+    focusCell(moveFocusTo);
+    event.preventDefault(); // prevent the page from moving with the arrow keys.
+  };
+
+  const onFocus = (event) => {
+    if (!event.currentTarget.contains(event.relatedTarget)) {
+      // Not triggered when swapping focus between children
+      focusCell(focusedCell.current);
```

Comment:
Good catch. In this commit I changed the focused cell data to row and column ids instead of indexes: https://github.com/cerner/terra-framework/pull/1919/commits/396ed597aeb86c4aad0c6304fa5c4f8f0e6b4f